### PR TITLE
DB-8947 add db2-like syscolumns and systables views to sysibm schema

### PR DESCRIPTION
**Pull request by yxia92**

### CODE REVIEW

**Comment by yxia92** [comment]
jenkins please test branch @cdh5.12.2

**Comment by spliceci** [comment]
TEST FAILED

**Comment by yxia92** [comment]
jenkins please test branch @cdh5.12.2

**Comment by spliceci** [comment]
TEST SUCCEEDED +1

File: splice_machine/src/main/java/com/splicemachine/derby/impl/sql/catalog/SpliceDataDictionary.java [hunk 1]
```diff
@@ -292,6 +292,49 @@ public void createSystemViews(TransactionController tc) throws StandardException
...
+
+        // drop it if it exists
+        if (td != null) {
+            ViewDescriptor vd = getViewDescriptor(td);
+
+            // drop the view deifnition
+            dropAllColumnDescriptors(td.getUUID(), tc);
+            dropViewDescriptor(vd, tc);
+            dropTableDescriptor(td, sysIBMSchemaDesc, tc);
+        }
```
Comment (jyuanca) [inline splice_machine/src/main/java/com/splicemachine/derby/impl/sql/catalog/SpliceDataDictionary.java]: Why sysibm.syscolumns and sysibm.systables can already exist?

File: splice_machine/src/test/java/com/splicemachine/derby/impl/sql/catalog/ViewsInSysIbmIT.java [hunk 2]
```diff
@@ -0,0 +1,224 @@
...
+                " F5  |  T5   |VIEWSINSYSIBMIT |-DECIMAL - |   Y   |    0    |  10   |   2   |   5   | DECIMAL  |    10     | NULL  |\n" +
+                " G5  |  T5   |VIEWSINSYSIBMIT |-TIMESTMP- |   N   |    0    |  10   |   6   |   6   |TIMESTAMP |    10     |   1   |\n" +
+                " H5  |  T5   |VIEWSINSYSIBMIT |-TIME    - |   Y   |    0    |   3   |   0   |   7   |  TIME    |     3     | NULL  |\n" +
+                " I5  |  T5   |VIEWSINSYSIBMIT |-CHAR    - |   Y   |  1252   |  10   |   0   |   8   |CHARACTER |    10     | NULL  |\n" +
+                " J5  |  T5   |VIEWSINSYSIBMIT |-VARCHAR - |   Y   |  1252   |  200  |   0   |   9   | VARCHAR  |    200    | NULL  |\n" +
+                " K5  |  T5   |VIEWSINSYSIBMIT |-BOOLEAN - |   Y   |    0    |   1   |   0   |  10   | BOOLEAN  |     1     | NULL  |\n" +
+                " L5  |  T5   |VIEWSINSYSIBMIT |-CLOB    - |   Y   |  1252   | 32765 |   0   |  11   |  CLOB    |   32765   | NULL  |\n" +
+                " M5  |  T5   |VIEWSINSYSIBMIT |-BLOB    - |   Y   |    0    |  -1   |   0   |  12   |  BLOB    |   32768   | NULL  |";
+
```
Comment (jyuanca) [inline splice_machine/src/test/java/com/splicemachine/derby/impl/sql/catalog/ViewsInSysIbmIT.java]: Is it possible that the result is not sorted by colno? If so, add a order by clause can avoid intermittent IT failures

File: splice_machine/src/main/java/com/splicemachine/derby/impl/sql/catalog/SpliceDataDictionary.java (hunk 1 above)
Comment (yxia92) [reply splice_machine/src/main/java/com/splicemachine/derby/impl/sql/catalog/SpliceDataDictionary.java]: I'm expecting that the views may need to be revised based on users feedback, so this function will be called to update the definition. Also for force upgrade scenarios, the views may also exist.

File: splice_machine/src/test/java/com/splicemachine/derby/impl/sql/catalog/ViewsInSysIbmIT.java (hunk 2 above)
Comment (yxia92) [reply splice_machine/src/test/java/com/splicemachine/derby/impl/sql/catalog/ViewsInSysIbmIT.java]: Yes, the result may not be sorted, but the assert statement will sort the result before comparing with the expected result:
```
        Assert.assertEquals("\n" + sqlText + "\n", expected, TestUtils.FormattedResult.ResultFactory.toString(rs));
```
So the order is deterministic.

**Pull request merged**